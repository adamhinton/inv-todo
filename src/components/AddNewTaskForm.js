import React from "react";
import styled from "styled-components";
import { useState } from "react";

const AddNewTaskForm = (props) => {
  const { todoList, setTodoList } = props;
  const [taskDescription, setTaskDescription] = useState("");
  //   {
  //     id: number,
  //     text: string,
  //     isCompleted: boolean,
  //   },

  return (
    <StyledForm
      onSubmit={(e) => {
        e.preventDefault();
        const nextId = todoList.length + 1;
        const taskObject = {
          id: nextId,
          text: taskDescription,
          completed: false,
        };

        setTodoList([...todoList, taskObject]);
        setTaskDescription("");
      }}
    >
      <StyledLabel>Add New Task</StyledLabel>
      <StyledInput
        data-testid="add-new-task-input"
        value={taskDescription}
        type="text"
        onChange={(e) => {
          setTaskDescription(e.target.value);
        }}
        placeholder="Enter Task Description..."
      ></StyledInput>
      <StyledButton type="submit">+ Add Task</StyledButton>
    </StyledForm>
  );
};

export default AddNewTaskForm;

const StyledForm = styled.form`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
`;

const StyledLabel = styled.label`
  width: 100%;
  margin-bottom: 3%;
  font-weight: bold;
`;

const StyledInput = styled.input`
  border: none;
  margin-left: 5%;
`;

const StyledButton = styled.button`
  border: none;
  background-color: #0d98ba;
  color: white;
  border-radius: 10px;
  padding: 2% 3%;
  font-size: 0.75rem;
`;
