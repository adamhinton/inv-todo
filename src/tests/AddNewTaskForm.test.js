import AddNewTaskForm from "../components/AddNewTaskForm";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

test("[1] AddNewTaskForm renders without errors", () => {
  render(<AddNewTaskForm />);
});

test("[2] Label, form field and add task button are visible", () => {
  render(<AddNewTaskForm />);

  const addNewTaskLabel = screen.getByText("Add New Task");
  const addNewTaskInput = screen.getByTestId("add-new-task-input");
  const addNewTaskButton = screen.getByText("+ Add Task");

  expect(addNewTaskLabel).toBeVisible();
  expect(addNewTaskInput).toBeVisible();
  expect(addNewTaskButton).toBeVisible();
});
