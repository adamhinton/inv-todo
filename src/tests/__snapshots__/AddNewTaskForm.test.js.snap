// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[3] Matches snapshot from 5.29.22 1`] = `
<form
  className="sc-bczRLJ jZCVrh"
  onSubmit={[Function]}
>
  <label
    className="sc-gsnTZi kgPqOe"
  >
    Add New Task
  </label>
  <input
    className="sc-dkzDqf fUOzCj"
    data-testid="add-new-task-input"
    onChange={[Function]}
    placeholder="Enter Task Description..."
    type="text"
    value=""
  />
  <button
    className="sc-hKMtZM fmPZnP"
    type="submit"
  >
    + Add Task
  </button>
</form>
`;
